classes-recipe/image_types_qcom.bbclass

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) 2023-2024 Qualcomm Innovation Center, Inc. All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+
+inherit image_types
+
+IMAGE_TYPES += "qcomflash"
+
+QCOM_ESP_IMAGE ?= "esp-qcom-image"
+QCOM_ESP_FILE ?= "${@'efi.bin' if d.getVar('QCOM_ESP_IMAGE') else ''}"
+IMAGE_QCOMFLASH_ESPIMG ?= "${DEPLOY_DIR_IMAGE}/${QCOM_ESP_IMAGE}-${MACHINE}.rootfs.vfat"
+
+# There is currently no upstream-compatible way for the firmware to
+# identify and load the correct DTB from a combined-dtb that contains all
+# dtbs defined in KERNEL_DEVICETREE, so pick the first individual image
+# generated by linux-qcom-dtbbin, unless a different one is specified.
+QCOM_DTB_DEFAULT ?= "${@os.path.basename(d.getVar('KERNEL_DEVICETREE').split()[0][:-4]) if d.getVar('KERNEL_DEVICETREE') else ''}"
+QCOM_DTB_FILE ?= "dtb.bin"
+IMAGE_QCOMFLASH_DTB ?= "${DEPLOY_DIR_IMAGE}/dtb-${QCOM_DTB_DEFAULT}-image.vfat"
+
+QCOM_ROOTFS_FILE ?= "system.img"
+IMAGE_QCOMFLASH_FS_TYPE ??= "ext4"
+IMAGE_QCOMFLASH_ROOTFS ?= "${IMGDEPLOYDIR}/${IMAGE_LINK_NAME}.${IMAGE_QCOMFLASH_FS_TYPE}"
+
+QCOMFLASH_DIR = "${WORKDIR}/qcomflash"
+IMAGE_CMD:qcomflash = "create_qcomflash_pkg"
+do_image_qcomflash[dirs] = "${QCOMFLASH_DIR}"
+do_image_qcomflash[cleandirs] = "${QCOMFLASH_DIR}"
+do_image_qcomflash[depends] += "qcom-gen-partition-bins:do_deploy virtual/kernel:do_deploy \
+				${@'${QCOM_ESP_IMAGE}:do_image_complete' if d.getVar('QCOM_ESP_IMAGE') != '' else  ''}"
+IMAGE_TYPEDEP:qcomflash += "${IMAGE_QCOMFLASH_FS_TYPE}"
+
+create_qcomflash_pkg() {
+    local oldpwd="$PWD"
```

Comment:
you don't need these 2 lines. bitbake will to cd QCOMFLASH_DIR for you, since you have it in [dirs]

Reply:
Yup, leftover, removed.